File: cli/internal/graph/testdata/swift/edge_cases.swift
import Foundation

// class NotARealClass — should be ignored (line comment)

/*
class AlsoNotReal
func alsoIgnoredFn() {}
*/

class Outer {
    /* nested block /* not really */ comment */
    func transform<T>(input: String) -> T {
        let literal = "class FakeClass { func fakeFn() {} }"
        let multi = """
            class TripleQuotedFake
            func anotherFake()
            """
        return input as! T
    }

    class Inner {
        func innerMethod() {}
    }
}

extension String {
    func extensionFn() -> Int { count }
}

public final class FinalClass {
    public static func staticFn() {}
    nonisolated func isolatedFn() {}
}

@MainActor
public func mainActorFn() {}

File: cli/internal/graph/testdata/kotlin/edge_cases.kt
package com.mobiai.demo

import com.x.Y

// class NotARealClass — should be ignored (line comment)

/*
class AlsoNotReal
fun alsoIgnoredFn() {}
*/

class Outer {
    /* nested block /* not really nested */ comment */
    private inline fun <T> transform(input: String): T {
        val literal = "class FakeClass { fun fakeFn() {} }"
        val multi = """
            class TripleQuotedFake
            fun anotherFake()
        """.trimIndent()
        return input as T
    }

    inner class Inner {
        fun innerMethod() {}
    }
}

fun String.extensionFn(): Int = length

suspend fun coroutineFn() {}

@Composable
public open fun ComposableScreen() {}
